### PR TITLE
ISPN-6869 Synchronize starting and stopping DefaultCacheManager

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/ISPN-6869

An approach based on 2 locks.

### CODE REVIEW

**Comment by vjuranek** [comment]
@slaskawi maybe I'm missing something, but this IMHO doesn't prevent deadlock described in ISPN-6869 -  AFAIKT new cache can be still started while cache manager stop is in progress and thus result into the deadlock.

File: core/src/main/java/org/infinispan/manager/DefaultCacheManager.java [hunk 1]
```diff
@@ -135,8 +135,8 @@
    private final CacheContainerStats stats;
    private final ConfigurationManager configurationManager;
 
-   @GuardedBy("this")
-   private boolean stopping;
+   private final Lock stopSemaphore = new ReentrantLock();
```
Comment (rvansa) [inline core/src/main/java/org/infinispan/manager/DefaultCacheManager.java]: semaphore?

Comment (slaskawi) [inline core/src/main/java/org/infinispan/manager/DefaultCacheManager.java]: Just a name... I can change it to lock :)

File: core/src/main/java/org/infinispan/manager/DefaultCacheManager.java [hunk 2]
```diff
@@ -687,18 +693,22 @@ private void terminate(String cacheName) {
    public void stop() {
       authzHelper.checkPermission(AuthorizationPermission.LIFECYCLE);
 
-      synchronized (this) {
-         if (stopping) {
-            log.trace("Ignore call to stop as the cache manager is stopping");
-            return;
+      //if start is in progress we need to wait until it's finished
+      startSemaphore.lock();
```
Comment (slaskawi) [inline core/src/main/java/org/infinispan/manager/DefaultCacheManager.java]: Here is a place where the code will block waiting for start to finish

Comment (vjuranek) [inline core/src/main/java/org/infinispan/manager/DefaultCacheManager.java]: well, but it's a lock for staring cache manager, not a cache - sorry, still unclear to me, how it helps

**Comment by slaskawi** [comment]
I was wrong here... It's a matter of stopping CacheManager and starting Caches (not CacheManager).. sorry for confusion!

**Pull request closed without merging**